### ISSUE TITLE
Refactor `checkApplicabilityPrecondition` so it can be used in driver where it doesn't raise error

### ISSUE THREAD

**Issue by balsa-sarenac** [guillep/pharo-core#15642]
Driver should be responsible to raise errors.
During scripting refactoring needs to raise error.
We need to create nice API that will be used by the driver to check the errors.

**Comment by Sahityaaryan** [comment]
can I work over this issue?

**Comment by balsa-sarenac** [comment]
@Sahityaaryan Of Course, let me know if you have any questions or clarifications on the architecture.

**Comment by Sahityaaryan** [comment]
@balsa-sarenac 

I think this has already been done, as the refactorings that have drivers are implemented using **failedApplicabilityPreconditions**. They simply provide a collection of failed conditions to the drivers, giving the drivers control over raising errors.  

## Why I am in dilemma  
The **checkApplicabilityPrecondition** method is used in the refactorings of the legacy implementation (RBs), which lack interaction drivers.  

## What I would have done  
I would try to create another implementation of **checkApplicabilityPrecondition** (to avoid breaking the legacy refactorings) that, instead of raising an error, provides either a message object (indicating what went wrong) to the corresponding driver or a list of unmatched conditions.  

However, it seems like this has already been done by **failedApplicabilityPreconditions**.  

If I am misunderstanding something, please correct me.

**Comment by balsa-sarenac** [comment]
@Sahityaaryan good points, now when I think about it we might not need to do this anymore. I'll close the issue. Thanks for looking into this!
